### PR TITLE
Reenabled Canton JVM metrics and prometheus filtering

### PR DESCRIPTION
**Pull request by daravep**
Before this change, we didn't have JVM metrics and we didn't have prometheus filtering. Now, we do.

Addresses https://github.com/DACH-NY/canton/issues/16647
and https://github.com/DACH-NY/canton/issues/17860
and https://github.com/DACH-NY/canton-network-node/issues/10440

### CODE REVIEW

File: canton/community/app-base/src/main/scala/com/digitalasset/canton/environment/Environment.scala [hunk 1]
```diff
@@ -9,14 +9,7 @@ import com.daml.grpc.adapter.ExecutionSequencerFactory
...
 import com.digitalasset.canton.config.*
-import com.digitalasset.canton.console.{
-  ConsoleEnvironment,
-  ConsoleGrpcAdminCommandRunner,
-  ConsoleOutput,
-  GrpcAdminCommandRunner,
-  HealthDumpGenerator,
-  StandardConsoleOutput,
-}
+import com.digitalasset.canton.console.{ConsoleEnvironment, ConsoleGrpcAdminCommandRunner, ConsoleOutput, GrpcAdminCommandRunner, HealthDumpGenerator, StandardConsoleOutput}
```
Comment (rgugliel-da) [inline canton/community/app-base/src/main/scala/com/digitalasset/canton/environment/Environment.scala:12]: Do you want to format to minimize future conflicts or you don't care ?

File: canton/community/app-base/src/main/scala/com/digitalasset/canton/metrics/CsvReporter.scala [hunk 2]
```diff
@@ -69,10 +60,10 @@ class CsvReporter(config: MetricsReporterConfig.Csv, val loggerFactory: NamedLog
     lock.synchronized {
       if (running.get()) {
         val ts = CantonTimestamp.now()
-        val filtered = metrics.asScala.filter(includeMetric).flatMap { data =>
```
Comment (rgugliel-da) [inline canton/community/app-base/src/main/scala/com/digitalasset/canton/metrics/CsvReporter.scala:72]: Can you explain why it is fine not to do the filtering anymore ?

Comment (rgugliel-da) [reply canton/community/app-base/src/main/scala/com/digitalasset/canton/metrics/CsvReporter.scala:72]: OK, it is done below

File: canton/community/app-base/src/main/scala/com/digitalasset/canton/metrics/MetricsRegistry.scala [hunk 3]
```diff
@@ -72,9 +97,21 @@ object MetricsReporterConfig {
       filters: Seq[MetricsFilterConfig] = Seq.empty,
   ) extends MetricsReporterConfig
 
+  /** Log metrics reporter configuration
+   *
+   * This reporter will log the metrics in the given interval
+   *
+   * @param interval how often to log the metrics
+   * @param filters which metrics to include
```
Comment (rgugliel-da) [inline canton/community/app-base/src/main/scala/com/digitalasset/canton/metrics/MetricsRegistry.scala:105]: Can you document what is the behavior of logAsInfo=false ?

File: canton/community/app-base/src/main/scala/com/digitalasset/canton/environment/Environment.scala (hunk 1 above)
Comment (daravep) [reply canton/community/app-base/src/main/scala/com/digitalasset/canton/environment/Environment.scala:12]: The plan is: 
- merge this PR
- create a snapshot
- update daml dependences in canton (will fail)
- apply this change as a patch
- reformat
- update codedrop

**Pull request merged**